src/main/java/org/jitsi/videobridge/Conference.java

```diff
@@ -1000,16 +982,11 @@ private Endpoint getEndpoint(String id, boolean create)
 
         synchronized (endpoints)
         {
-            for (Iterator<Endpoint> i = endpoints.iterator(); i.hasNext();)
-            {
-                Endpoint e = i.next();
+            endpoints.removeIf(Endpoint::isExpired);
 
-                if (e.isExpired())
-                {
-                    i.remove();
-                    changed = true;
-                }
-                else if (e.getID().equals(id))
+            for (Endpoint e : endpoints)
```

Comment:
if we want to continue the java 8-ness, this could be:

```
endpoint = endpoints.stream()
    .filter(e -> e.getID().equals(id))
    .findFirst()
    .orElse(null);
```

Reply:
In pure Java 8-ness, the return should be `Optional<Endpoint>` instead of `void`.

Reply:
I didn't understand that last part, and I'd rather avoid changing too much. I adopted Brian's suggestion.

Reply:
[Javadoc](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html) or some [(anti) usage examples](https://dzone.com/articles/optional-ispresent-is-bad-for-you)

Reply:
This looks consistent with the article. 

Or are you suggesting to change the return type of getEndpoint to Optional<Endpoint>? I'll leave this as it is for now, but I will also think about where it's appropriate to use Optional. Thanks for bringing it up!

Reply:
Yes, I meant changing the return type. I'm not really a fan of `Optional`, just wanted to bring it up. It's more important to return empty lists/arrays instead of null at other places.